trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/RequestBuilder.java

```diff
@@ -44,7 +45,7 @@ private RequestBuilder() {}
 
     static Request fromRequest(ContainerRequest request)
     {
-        MultivaluedMap<String, String> requestHeaders = request.getHeaders();
+        MultiMap<String> requestHeaders = ImmutableMultiMap.copyOf(request.getHeaders().entrySet(), false);
```

Comment:
Use the version of `copyOf` that takes only a `MultiMap` to take advantage of no-cost copying.